deepspeed/launcher/launch.py

```diff
@@ -219,7 +223,13 @@ def main():
                 cmd.append(f"--local_rank={local_rank}")
             cmd += args.training_script_args
 
-            process = subprocess.Popen(cmd, env=current_env)
+            if args.enable_each_rank_log:
+                date_str = time.strftime("%Y%m%d%H%M%S", time.localtime())
+                log_file = os.path.join(PID_FILE_BASEPATH, f"{date_str}_rank{local_rank}.log")
```

Comment:
Really love this PR, thank you @guoyejun! Could I request two small changes though? Let's allow the log file path to be user configurable. Perhaps that path can be an argument to `--enable_each_rank_log`?

Also, can you use `dist_rank` in the filename instead of `local_rank`? This would allow someone to write out the log files to a shared filesystem and still be able to distinguish between global ranks in multi-node deployments.

Reply:
sure, nice idea, will update the PR.